File: ws/chiriboga/u1/ws08CDJavaClasses/ChickenFarmSimulator/src/ec/edu/espe/farmsimulator/model/ChickenCoop.java
package ec.edu.espe.farmsimulator.model;

import java.util.List;

/**
 *
 * @autor Kerlly Chiriboga, ODS
 */
public class ChickenCoop {
    private int id;
    private List<Chicken> chickens;

    public ChickenCoop(int id, List<Chicken> chickens) {
        this.id = id;
        this.chickens = chickens;
    }

    @Override
    public String toString() {
        StringBuilder chickensInfo = new StringBuilder();
        for (Chicken chicken : chickens) {
            chickensInfo.append(chicken.toString()).append("\n");
        }
        return "ChickenCoop {" + id + "} \n" + chickensInfo.toString();
    }
    
    public String toCSV() {
        StringBuilder csvData = new StringBuilder();
        csvData.append("Coop ID: ").append(id).append("\n");
        for (Chicken chicken : chickens) {
            csvData.append(chicken.toCSV()).append("\n");
        }
        return csvData.toString();
    }

    public void addChicken(Chicken chicken) {
        chickens.add(chicken);
    }

    public void removeChicken(Chicken chicken) {
        chickens.remove(chicken);
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public List<Chicken> getChickens() {
        return chickens;
    }

    public void setChickens(List<Chicken> chickens) {
        this.chickens = chickens;
    }
}

File: ws/chiriboga/u1/ws08CDJavaClasses/ChickenFarmSimulatorPython/farm_simulator/model/ChickenCoop.py
# File: ec/edu/espe/farmsimulator/model/ChickenCoop.py

from farm_simulator.model import Chicken

class ChickenCoop:
    def __init__(self, id, chickens):
        self.id = id
        self.chickens = chickens

    def __str__(self):
        chickens_info = "\n".join([str(chicken) for chicken in self.chickens])
        return f"ChickenCoop {{ {self.id} }} \n{chickens_info}"

    def to_csv(self):
        csv_data = f"Coop ID: {self.id}\n"
        csv_data += "\n".join([chicken.to_csv() for chicken in self.chickens])
        return csv_data

    def add_chicken(self, chicken):
        self.chickens.append(chicken)

    def remove_chicken(self, chicken):
        self.chickens.remove(chicken)

